src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAzureUtilityImpl.java

```diff
@@ -0,0 +1,240 @@
+package ca.corefacility.bioinformatics.irida.repositories.filesystem;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.GZIPInputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+import ca.corefacility.bioinformatics.irida.util.FileUtils;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.blob.models.BlobStorageException;
+
+/**
+ * Component implementation of file utitlities for azure storage
+ */
+@Component
+public class IridaFileStorageAzureUtilityImpl implements IridaFileStorageUtility {
+	private static final Logger logger = LoggerFactory.getLogger(IridaFileStorageAzureUtilityImpl.class);
+
+	private BlobServiceClient blobServiceClient;
+	private BlobContainerClient containerClient ;
+
+	@Autowired
+	public IridaFileStorageAzureUtilityImpl(String connectionStr, String containerName){
+		this.blobServiceClient = new BlobServiceClientBuilder().connectionString(connectionStr)
+				.buildClient();
+		this.containerClient = blobServiceClient.getBlobContainerClient(containerName);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public File getTemporaryFile(Path file) {
+		File fileToProcess = null;
+
+		// We set the blobClient "path" to which file we want to get
+		BlobClient blobClient = containerClient.getBlobClient(getAzureFileAbsolutePath(file));
+
+		try {
+			InputStream initialStream = blobClient.openInputStream();
+			File targetFile = new File(file.toAbsolutePath().toString());
+			org.apache.commons.io.FileUtils.copyInputStreamToFile(initialStream, targetFile);
+			initialStream.close();
+			fileToProcess = targetFile;
```

Comment:
As discussed lets try switching this to use `Files.createTempFile` instead of downloading to this specified location.

Reply:
Updated in bdee3fd5ecbe7c0e5096d3f5889a51af59eac4f2